packages/dds/tree/src/changeset/delta.ts

```diff
@@ -0,0 +1,313 @@
+ */
+export interface ModifyMovedOut {
+    type: typeof MarkType.Modify;
+    setValue?: Value;
+    fields?: FieldMarks<ModifyMovedOut | Delete | MoveOut>;
+}
+
+/**
+ * Describes modifications made to a subtree that is being moved in.
+ */
+export interface ModifyMovedIn {
+    type: typeof MarkType.Modify;
+    fields?: FieldMarks<ModifyMovedIn | MoveIn | Insert>;
+}
+
+/**
+ * Describes modifications made to a subtree that is being inserted.
+ */
+export interface ModifyInserted {
+    type: typeof MarkType.Modify;
+    fields?: FieldMarks<ModifyMovedIn | MoveIn>;
+}
+
+/**
+ * Describes the deletion of a contiguous range of node.
+ * Includes descriptions of the modifications made to those nodes (if any).
+ */
+export interface Delete {
+    type: typeof MarkType.Delete;
+    count: number;
+    modify?: PositionedMarks<ModifyDeleted>;
+}
+
+/**
+ * Describes the moving out of a contiguous range of node.
+ * Includes descriptions of the modifications made to those nodes (if any).
+ */
+export interface MoveOut {
+    type: typeof MarkType.MoveOut;
+    count: number;
+    /**
+     * The delta should carry exactly one `MoveIn` mark with the same move ID.
+     */
+    moveId: MoveId;
+    modify?: PositionedMarks<ModifyMovedOut>;
+}
+
+/**
+ * Describes the moving in of a contiguous range of node.
+ * Includes descriptions of the modifications made to those nodes (if any).
+ */
+export interface MoveIn {
+    type: typeof MarkType.MoveIn;
+    /**
+     * The delta should carry exactly one `MoveOut` mark with the same move ID.
+     */
+    moveId: MoveId;
+    modify?: PositionedMarks<ModifyMovedIn>;
+}
+
+/**
+ * Describes the insertion of a contiguous range of node.
+ * Includes descriptions of the modifications made to those nodes (if any).
+ * Those are represented as `MoveIn` marks within `ProtoField`s.
+ */
+export interface Insert {
+    type: typeof MarkType.Insert;
+    content: ProtoNode[];
+    modify?: PositionedMarks<ModifyInserted>;
+}
+
+/**
+ * The contents of a subtree to be created
+ */
+export interface ProtoNode {
+    id?: NodeId;
+    type?: string;
+    value?: Value;
+    /**
+     * The fields of a subtree to be created
+     */
+    fields?: FieldMap<ProtoField>;
+}
+
+export type ProtoField = ProtoNode[];
+
+/**
+ * Uniquely identifies a MoveOut/MoveIn pair within a transaction.
+ * When the delta represents a whole transaction the MoveIn and MoveOut marks will both appear in the same delta.
+ * Deltas can also represent a portion of a transaction, in which case it's possible the MoveIn and MoveOut marks with
+ * the same `MoveId` will appear in different deltas.
+ */
+export type MoveId = Opaque<Brand<number, "delta.MoveId">>;
+
+export type Offset = number;
+
+/**
+ * An identifier that a node might carry.
+ * No uniqueness guarantees (across any scope) are made at this time.
+ * TODO: Update comment once uniqueness guarantees can be made.
+ */
+export type NodeId = Opaque<Brand<string, "delta.NodeId">>;
```

Comment:
The pattern for this has changed recently to get better tooling support (you might want to merge main into your branch to see the docs for it).

```suggestion
export interface NodeId extends Opaque<Brand<string, "delta.NodeId">>{}
```

Reply:
Same for MoveId above